linera-core/src/worker.rs

```diff
@@ -1062,6 +1061,11 @@ where
         }
         let local_time = self.storage.current_time();
         let outcome = chain.execute_block(block, local_time).await?;
+        // Check if tip_state would be valid.
+        chain
+            .tip_state
+            .get()
+            .verify_block_chaining(block, &outcome)?;
```

Comment:
I think we _should_ check block chaining (i.e. that it's at the expected height with the expected parent) before performing the potentially expensive execution.

So I'd really move those arithmetic checks out of `verify_block_chaining` and just do them directly here.